cpp/src/arrow/compute/row/compare_internal.cc

```diff
@@ -276,7 +276,8 @@ void KeyCompare::CompareVarBinaryColumnToRowHelper(
       int32_t tail_length = length - j * 8;
       uint64_t tail_mask = ~0ULL >> (64 - 8 * tail_length);
       uint64_t key_left = 0;
-      std::memcpy(&key_left, key_left_ptr + j, tail_length);
+      const uint8_t* src_bytes = reinterpret_cast<const uint8_t*>(key_left_ptr + j);
```

Comment:
As explained in https://github.com/apache/arrow/pull/46124#issuecomment-2814416656 , it could be confusing to people why this casting is ever needed, given that it is eventually passed down into `std::memcpy` as a `void *`. We can use a comment.
```suggestion
      // NOTE: UBSAN may falsely report "misaligned load" in `std::memcpy` on some
      // platforms when using 64-bit pointers. Cast to an 8-bit pointer to work around
      // this.
      const uint8_t* src_bytes = reinterpret_cast<const uint8_t*>(key_left_ptr + j);
```

Reply:
Thanks